.github/workflows/docs.yml

```diff
@@ -25,11 +25,8 @@ jobs:
       - name: Set up cache
         uses: DeterminateSystems/magic-nix-cache-action@main
 
-      - name: Checkout repository
-        uses: actions/checkout@v4
-
       - name: Build docs
-        run: nix -L build .#docs
+        run: nix -L build github:${{ github.repository }}/${{ github.sha }}#docs
```

Comment:
Does using this redundantly download and install the repository that is already available to the GitHub runner?

Reply:
I removed the `actions/checkout` step, so as far as I know, the repository shouldn't be downloaded to the working directory at all.

Reply:
If that's the case, then it should not be redundant. Nonetheless, it might be safe to assume that GitHub runners download the repository faster from their own server than from an external one. Maybe `actions/checkout` internally even `cp` the repository into the runner when using `actions/checkout`. IMHO it might be safer to assume their internal tooling to be better.

Reply:
I would assume it uses normal API calls since it needs to work on self-hosted runners too.

Either way, I thought this might be better from a testing standpoint since it downloads Stylix the same way a real user would do it, so it can catch any impurities which require the repository to be cloned directly. Of course, these should be prevented by Nix regardless.

If we want to optimize CI performance any further, the time taken to download and install Nix itself has a much bigger impact than this, especially on Darwin.

Reply:
> If we want to optimize CI performance any further, the time taken to download and install Nix itself has a much bigger impact than this, especially on Darwin.

Agreed. Probably not worth any more effort.

> I thought this might be better from a testing standpoint since it downloads Stylix the same way a real user would do it, so it can catch any impurities which require the repository to be cloned directly. Of course, these should be prevented by Nix regardless.

Good idea! Any way we could integrate this into `nix flake check`?